Remove duplicate macros between j9nonbuilder.h and j9consts.h

AidanHa opened the pull request:
-There were various duplicate macros in j9consts.h and j9nonbuilder.h. Since j9nonbuilder.h seemed to be more up to date, I removed several constants from j9consts.h, including the following:

Note: This is the second pull request attempt.

J9_JAVA_PUBLIC
J9_JAVA_PRIVATE
J9_JAVA_PROTECTED 0x4
J9_JAVA_STATIC (CARE IN STACK WALKER.JAVA)
J9_JAVA_FINAL
J9_JAVA_CLASS_RAM_SHAPE_SHIFT
J9_JAVA_CLASS_REFERENCE_SHIFT
J9_JAVA_SYNC
J9_JAVA_NATIVE
J9_JAVA_GETTER_METHOD
J9_JAVA_INTERFACE
J9_JAVA_UNUSED_0X200
J9_JAVA_ABSTRACT
J9_JAVA_FORWARDER_METHOD
J9_JAVA_EMPTY_METHOD
J9_JAVA_CLASS_DEPTH_MASK
J9_JAVA_CLASS_ARRAY
J9_JAVA_CLASS_RAM_ARRAY
J9_JAVA_METHOD_VTABLE
J9_JAVA_METHOD_HAS_EXCEPTIONS
J9_JAVA_CLASS_PRIMITIVE_TYPE
J9_JAVA_CLASS_UNSAFE
J9_JAVA_METHOD_FRAME_ITERATOR_SKIP
J9_JAVA_METHOD_CALLER_SENSITIVE
J9_JAVA_CLASS_BYTECODES_MODIFIED
J9_JAVA_CLASS_OWNABLE_SYNCHRONIZER
J9_JAVA_CLASS_HAS_EMPTY_FINALIZE
J9_JAVA_CLASS_HAS_JDBC_NATIVES
J9_JAVA_METHOD_OBJECT_CONSTRUCTOR
J9_JAVA_CLASS_GC_SPECIAL
J9_JAVA_CLASS_HAS_VERIFY_DATA
J9_JAVA_METHOD_UNUSED_1000000
J9_JSVS_CLASS_CONTENDED
J9_JAVA_CLASS_HAS_FINAL_FIELDS
J9_JAVA_METHOD_HAS_GENERIC_SIGNATURE
J9_JAVA_CLASS_HOT_SWAPPED_OUT
J9_JAVA_CLASS_HAS_CLINIT
J9_JAVA_CLASS_DYING
J9_JAVA_CLASS_REFERENCE_WEAK
J9_JAVA_FIELD_UNUSED_10000000
J9_JAVA_CLASS_REFERENCE_SOFT
J9_JAVA_CLASS_REFERENCE_MASK
J9_JAVA_CLASS_REFERENCE_PHANTOM
J9_JAVA_CLASS_CLONEABLE
J9_JAVA_CLASS_ROMRAMMASK

-I renamed the constants in various files. I did not touch anything in any of the DDR files @tajila
-There were alot of constants to renamed, so hopefully I did not made any typos.
I also added <#include "j9nonbuilder.h"> where necessary.

I also updated the Copyright comments where necessary, and modified the .gitignore file to ignore any .vscode folders

AidanHa commented:
Ah, thank you for the link @fjeremic I will edit the commit messages.

The multiple unnecessary sign offs come from me squashing multiple smaller commits together. Will I only need to sign off one commit in this case (sign off the latest commit)?

fjeremic commented:
> Will I only need to sign off one commit in this case (sign off the latest commit)?

That's correct.

tajila commented on .gitignore:
```diff
@@ -1,4 +1,4 @@
-# Copyright (c) 2000, 2018 IBM Corp. and others
+# Copyright (c) 2000, 2019 IBM Corp. and others
 #
```
can you please revert the changes in this file

tajila commented on runtime/compiler/env/VMJ9.cpp:
```diff
@@ -32,6 +32,7 @@
 #include "cache.h"
 #include "fastJNI.h"
 #include "j9cfg.h"
+
```
please remove this added whitespace

tajila commented on runtime/compiler/il/symbol/J9Symbol.cpp:
```diff
@@ -35,6 +35,7 @@
 #include <stddef.h>                            // for size_t
 #include <stdint.h>                            // for uint32_t, int32_t, etc
 #include <string.h>                            // for memcmp, strncmp
+
```
same here

tajila commented on runtime/compiler/runtime/SymbolValidationManager.hpp:
```diff
@@ -32,6 +32,7 @@
 #include "env/jittypes.h"                  // for uintptrj_t
 #include "j9.h"
 #include "j9nonbuilder.h"
+#include "j9consts.h"
```
was this needed in this here

tajila commented on runtime/gc_base/ClassLoaderManager.hpp:
```diff
@@ -32,7 +32,6 @@
 #include "j9.h"
 #include "j9cfg.h"
 
-
```
undo this change

tajila commented on runtime/gc_glue_java/ConcurrentMarkingDelegate.hpp:
```diff
@@ -33,6 +33,7 @@
 #include "j9modron.h"
 #include "j9nongenerated.h"
 #include "j9nonbuilder.h"
+#include "j9consts.h"
```
what was this needed for?

tajila commented on runtime/gc_glue_java/MarkingDelegate.hpp:
```diff
@@ -25,6 +25,7 @@
 #define MARKINGDELEGATE_HPP_
 
 #include "j9nonbuilder.h"
+#include "j9consts.h"
```
same here

tajila commented on runtime/gc_glue_java/ObjectModelDelegate.hpp:
```diff
@@ -24,6 +24,7 @@
 #define OBJECTMODELDELEGATE_HPP_
 
 #include "j9nonbuilder.h"
+#include "j9consts.h"
```
^^

tajila commented on runtime/oti/j9nonbuilder.h:
```diff
@@ -33,6 +33,7 @@
 #include "j9vmconstantpool.h"
 #include "j9consts.h"
 #include "jvmti.h"
+#include "omrsrp.h"
```
is this still needed here, now that the macros are moved over to j9const.h ?

tajila commented on runtime/vm/guardedstorage.c:
```diff
@@ -24,6 +24,7 @@
 #include "j9gs.h"
 #include "j9port_generated.h"
 #include "j9nonbuilder.h"
+#include "j9consts.h"
```
is this needed?

AidanHa commented:
@tajila for the some of the #include "j9consts.h" calls, I just wanted to make sure that all files that included nonbuilder.h also had j9consts (kind of a surefire way to make sure there were no errors). I will now remove all the unessesary #includes.

tajila commented on sourcetools/j9constantpool/com/ibm/oti/VMCPTool/ConstantPoolStream.java:
```diff
@@ -388,4 +388,4 @@ public void comment(String string) {
 		}
 	}
-}
+}
```
neeed to keep the newline

tajila commented:
One minor comment rest LGTM

@dmitripivkine Please review the GC changes

@fjeremic Please review the JIT changes

keithc-ca commented on debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/j9/gc/GCClassArrayClassSlotIterator.java:
```diff
@@ -47,7 +48,7 @@ protected GCClassArrayClassSlotIterator(J9ClassPointer clazz) throws CorruptData
 			addresses.add(VoidPointer.cast(clazz.arrayClassEA()));
 		}
-		if(clazz.romClass().modifiers().allBitsIn(J9_JAVA_CLASS_ARRAY)) {
+		if(clazz.romClass().modifiers().allBitsIn(J9AccClassArray)) {
```
This will break the ability to look at core files generated before this change.

AidanHa replied:
Should I be reverting the macro name change here? Or is there a better fix to this problem?

keithc-ca replied:
I spoke with @tajila who assures me that the values in the two classes are equal and so this should be compatible with older core files; I'll resume my review...

keithc-ca commented:
> > Will I only need to sign off one commit in this case (sign off the latest commit)?
> 
> That's correct.

No, each commit needs a `Signed-off-by:` line.

keithc-ca commented on debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/j9/gc/GCClassArrayClassSlotIterator.java:
```diff
@@ -30,6 +30,7 @@
 import com.ibm.j9ddr.vm29.pointer.generated.J9ClassPointer;
 
 import static com.ibm.j9ddr.vm29.structure.J9Consts.*;
```
I believe this import becomes unused: please remove it and similar imports elsewhere.

keithc-ca commented on debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/j9/gc/GCClassArrayClassSlotIterator.java:
```diff
@@ -30,6 +30,7 @@
 import com.ibm.j9ddr.vm29.pointer.generated.J9ClassPointer;
 
 import static com.ibm.j9ddr.vm29.structure.J9Consts.*;
+import static com.ibm.j9ddr.vm29.structure.J9JavaAccessFlags.*;
```
This file only uses one field of that class; it would be clearer to say:
```
import static com.ibm.j9ddr.vm29.structure.J9JavaAccessFlags.J9AccClassArray;
```

AidanHa commented:
Sure, let me add the fixes and do a rebase to see if anything will change

AidanHa commented:
I've rebased this branch to Eclipse:master just over an hour ago, and ran a build on my local machine (which ended up being successful) @keithc-ca. I've made changes to address the comments above.

keithc-ca commented:
Jenkins test sanity zlinux,win jdk8,jdk11

gacholio commented:
Reverted this due to DDR compile failures in the internal builds.

keithc-ca commented:
@AidanHa This might work in reverse: removing uses of `J9JavaAccessFlags` in favour of using constants in `J9Consts`.

AidanHa commented:
I'm not too familiar with the DDR code but from my point of view, I noticed there are no macros in `j9consts.h` that map to the ddr namespace (in other words, no there’s no code like
`/* @DDR_namespace: map_to_type=J9JavaAccessFlags */` in `j9consts.h`). Perhaps when the DDR is being compiled, then it doesn't even look at `j9consts.h`? If that is the case, where can I go to fix this?

keithc-ca commented:
If there were no internal IBM builds, your change would have been fine. If we're going to consider eliminating duplicate constants, we must use only those that also *were* available in the past so that old core files produced by IBM builds can be analyzed.

gacholio commented:
While I agree that's good practice, there will be circumstances where the back compatibility pain will be too high. You can always use the DDR that shipped with the drop that produced the core to analyze it.

AidanHa commented:
![image](https://user-images.githubusercontent.com/33072072/52657634-1bb92980-2ec7-11e9-8932-64f60932aa7f.png)
We were able to use the new solution, and it successfully produced a class file S|J9JavaAccessFlags|J9JavaAccessFlagsPointer| for vm 28.  @keithc-ca, what should the next step be? 

The screenshot above is from the superset file, and it contains all the macros that we moved from j9nonbuilder.h

AidanHa commented:
For context, I made a new file `j9javaaccessflags.h`, moved the macros into that file, and changed the `vmddrstructs.properties` file to build the macros from the new file.
